### PR TITLE
Traffic light detection - dummy mode

### PR DESCRIPTION
**Pull request by itornaza**
## What?
Created a dummy detector for the traffic lights using the /vehicle/traffic_lights topic that the simulator publishes to.

## Why?
The /traffic_waypoint is published and further development can be done until the live detector is implemented.

[x] - works on simulator
[x] - tested locally
[ ] - local env version: Simulator running on OS X 10.13.3 and ros running on VirtualBox

### CODE REVIEW

**Review by fadida — APPROVED** [review]
Look good...
I have one question though, why not use inheritance and build an abstract detector and make both dummy and live detectors derive from it?

I think that it will make it easier to maintain this code without duplicating functionality.

**Comment by itornaza** [comment]
Hi @fadida! Thank you for the comment. Using a detector base class is a great suggestion! I have not include that yet because I wanted to get through the requirements as quickly as possible in order to facilitate the development for all the other project nodes. In my opinion, we can merge this branch as is and implement inheritance while we build the `live_detector.py` after @ooleksyuk completes her classifier on a new branch. Does this approach make sense?

**Review by ooleksyuk — APPROVED** [review]
LGTM.

**Review by jae1e — APPROVED** [review]

File: ros/src/tl_detector/dummy_detector.py [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+            # Correlate each light position (x, y) with the closest stop line
+            x = light.pose.pose.position.x
+            y = light.pose.pose.position.y
+            self.light_stops[(x, y)] = self.stop_line_positions[matching_index]
+
+            rospy.loginfo("light = (%s, %s) : stop = (%s, %s)",
+                          x, y, self.stop_line_positions[matching_index][0],
+                          self.stop_line_positions[matching_index][1])
+
+    def get_closest_waypoint(self, x, y):
```
Comment (isharaka) [inline ros/src/tl_detector/dummy_detector.py:167]: get_closeset_waypoint is currently O(n) and this is called at every traffic callback. On my machine this took over 8ms. This is mostly critical for live tracking. But even with simulated tracking we see discernible improvement when using a tree to search.

**Review by isharaka — APPROVED** [review]

File: ros/src/tl_detector/dummy_detector.py (hunk 1 above)
Comment (itornaza) [reply ros/src/tl_detector/dummy_detector.py:167]: Shall we merge this and then call your tree search function from the dummy?

**Comment by itornaza** [comment]
Note that you can switch from dummy to live detector from the /ros/src/tl_detector/tl_detector.py

**Pull request merged**